core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerIterator.java

```diff
@@ -58,18 +59,24 @@
   private long batchCount = 0;
   private long readaheadThreshold;
 
+  private Set<ScannerIterator> activeIters;
+
   private static ThreadPoolExecutor readaheadPool = new ThreadPoolExecutor(0, Integer.MAX_VALUE, 3L,
       TimeUnit.SECONDS, new SynchronousQueue<>(),
       new NamingThreadFactory("Accumulo scanner read ahead thread"));
 
+  private boolean closed = false;
+
   ScannerIterator(ClientContext context, Table.ID tableId, Authorizations authorizations,
       Range range, int size, long timeOut, ScannerOptions options, boolean isolated,
-      long readaheadThreshold) {
+      long readaheadThreshold, Set<ScannerIterator> activeIters) {
```

Comment:
I see you are using a SynchronizedSet in ScannerImpl but this constructor only takes a set.  Isn't it possible to pass in a non synchronized set?

Reply:
I changed how this was implemented.